cropduster/utils/gifsicle.py

```diff
@@ -19,11 +19,9 @@ def __init__(self, im):
         self.pil_image = im
         self.size = im.size
 
-        filename = getattr(im, 'filename', None)
-        if not filename or not os.path.exists(filename):
-            temp_file = tempfile.NamedTemporaryFile(suffix='.gif')
-            filename = temp_file.name
-            im.save(filename)
+        temp_file = tempfile.NamedTemporaryFile(suffix='.gif', delete=False)
+        filename = temp_file.name
+        im.save(filename)
```

Comment:
This temp file won't get deleted. But you don't actually need to create a temporary file here, you could use a `BytesIO` object and pass it as stdin to the gifsicle process. So the relevant methods (`__init__` and `save`) could be simplified to:

```python
class GifsicleImage(object):

    def __init__(self, im):
        # ...
        self.src_buf = BytesIO()
        im.save(self.src_buf, format=im.format)
        # ...

    def save(self, buf, **kwargs):
        self.src_buf.seek(0)
        proc = subprocess.Popen(self.args,
            stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        out, err = proc.communicate(input=self.src_buf)
        logger.debug(err)
        buf.write(out)
        buf.seek(0)
```

Reply:
This more or less worked 👍 As you can see, though, I had to make some updates to ensure individual frame durations were retained- https://github.com/theatlantic/django-cropduster/pull/61/files#diff-2bb695ab48526ef1eb8b528c89cdbfb3R29